quantum/plugins/ibm/compiler/QObjectCompiler.cpp

```diff
@@ -107,9 +107,11 @@ QObjectCompiler::translate(std::shared_ptr<xacc::CompositeInstruction> function)
   std::vector<xacc::ibm::Experiment> experiments;
 
   auto uniqueBits = function->uniqueBits();
-
+  // The number of qubits required for an experiment is the number of *physical* qubits,
+  // i.e. the max index of qubit used in the circuit.  
+  auto nbRequiredBits = function->nPhysicalBits();
   auto visitor =
-      std::make_shared<QObjectExperimentVisitor>(function->name(), uniqueBits.size());
+      std::make_shared<QObjectExperimentVisitor>(function->name(), nbRequiredBits);
```

Comment:
This came up with Tyler yesterday and this was the fix we put in. I'll accept yours and tell him to pull the latest from master.

Reply:
Yea, I've also discussed w/ Tyler a while ago but didn't see a pending PR hence went ahead w/ it :)